src/main/java/seedu/fast/logic/parser/AppointmentCommandParser.java

```diff
@@ -0,0 +1,66 @@
+package seedu.fast.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.fast.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.fast.logic.parser.CliSyntax.PREFIX_APPOINTMENT;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import seedu.fast.commons.core.index.Index;
+import seedu.fast.commons.exceptions.IllegalValueException;
+import seedu.fast.logic.commands.AppointmentCommand;
+import seedu.fast.logic.parser.exceptions.ParseException;
+import seedu.fast.model.person.Appointment;
+
+public class AppointmentCommandParser implements Parser {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AppointmentCommand
+     * and returns an AppointmentCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AppointmentCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_APPOINTMENT);
+
+        Index index;
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (IllegalValueException ive) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AppointmentCommand.MESSAGE_USAGE), ive);
+        }
+
+        String date = argMultimap.getValue(PREFIX_APPOINTMENT).orElse(Appointment.NO_APPOINTMENT);
+        date = parseDateString(date);
```

Comment:
Minor nitpick, might want to name the parsed date as a new variable (e.g. parsedDate) to minimize confusion for people reading through the code

Reply:
Yes, I agree that the word `appointment` is pretty long. 
Since we have a user stories #27 that allows the use of friendlier syntax, we can resolve this issue in future implementation, so that the usage of friendlier syntax will be more standardized and consistent for the entire program.